rpcs3/rpcs3qt/log_frame.cpp

```diff
@@ -430,41 +430,41 @@ void log_frame::UpdateUI()
 {
 	const auto start = steady_clock::now();
 
+	const auto tty = g_fxo->get<tty_t>();
+
 	// Check TTY logs
-	while (const u64 size = std::max<s64>(0, g_tty_size.load() - (m_tty_file ? m_tty_file.pos() : 0)))
+	if (tty && tty->fd && m_tty_time != tty->time)
 	{
-		std::string buf;
-		buf.resize(size);
-		buf.resize(m_tty_file.read(&buf.front(), buf.size()));
+		// save old selection
+		QTextCursor text_cursor{ m_tty->document() };
+		const int sel_pos = text_cursor.position();
+		int sel_start = text_cursor.selectionStart();
+		int sel_end = text_cursor.selectionEnd();
+
+		m_tty->clear();
```

Comment:
why the heck are you clearing the text edit.....

Reply:
Appending text is not the only thing it does now, text can be changed enturiely between different boots.

Reply:
This is bullshit. might as well clear the entire log between boots.
Also, it's slow as fuck. so you might as well remove all your improvements since it's completely working against each other.

Reply:
Fixed the optimization flaw, thanks for mentioning it.
As a person who uses the TTY regularly, interleaving text between different boots of differnet is very confusing. Unlike rpcs3.log which is ver explicit about where each boot starts and the earlier finishes, TTY on the other hand has no such explicit separations.
This is especially triggering when the older TTY output did not contain a '\n' at the end of it resulting in soemthing like:

Old boot: "This is TTY of TEST12345"
New boot: "This is the TTY of 12345TEST"
Combined: "This is TTY of TEST12345This is the TTY of 12345TEST"

Can you see how this may be confusing for some people?

Reply:
Just log some header to the tty on boot. What's the problem?

It's much worse to force a clear upon users